packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -391,37 +411,115 @@ jsi::Value createJSRuntimeError(
       .call(runtime, message);
 }
 
+/**
+ * Cpp struct to hold JVMFrame
+ */
+struct SimpleJFrame {
+    std::string className;
+    std::string fileName;
+    int lineNumber;
+    std::string methodName;
+};
+
+/**
+ * Cpp struct to hold JThrowable
+ */
+struct SimpleJThrowable {
+    std::string name;
+    std::string message;
+    std::vector<SimpleJFrame> stackElements;
+};
+
+/**
+ * Extracts throwable properties to JVM env independent data holder.
+ */
+SimpleJThrowable extractThrowable(
+            jni::local_ref<jni::JThrowable> throwable) {
+    SimpleJThrowable extracted;
+    auto stackTrace = throwable->getStackTrace();
+
+    extracted.stackElements.reserve(stackTrace->size());
+    for (int i = 0; i < stackTrace->size(); ++i) {
+        auto frame = stackTrace->getElement(i);
+        extracted.stackElements.push_back(SimpleJFrame {
+                frame->getClassName(),
+                frame->getFileName(),
+                frame->getLineNumber(),
+                frame->getMethodName()
+        });
+    }
+
+    extracted.name = throwable->getClass()->getCanonicalName()->toStdString();
+    extracted.message = throwable->getMessage()->toStdString();
+
+    return extracted;
+}
+
+/**
+ * Converts simple JVM independent Throwable container to JSError.
+ */
+jsi::JSError convertSimpleJThrowableToJSError(
+        jsi::Runtime& runtime,
+        SimpleJThrowable throwable) {
+    auto stackTrace = throwable.stackElements;
+
+    jsi::Array stackElements(runtime, stackTrace.size());
+    for (int i = 0; i < stackTrace.size(); ++i) {
+        auto frame = stackTrace[i];
+
+        jsi::Object frameObject(runtime);
+        frameObject.setProperty(runtime, "className", frame.className);
+        frameObject.setProperty(runtime, "fileName", frame.fileName);
+        frameObject.setProperty(runtime, "lineNumber", frame.lineNumber);
+        frameObject.setProperty(runtime, "methodName", frame.methodName);
+        stackElements.setValueAtIndex(runtime, i, std::move(frameObject));
+    }
+
+    jsi::Object cause(runtime);
+    auto name = throwable.name;
+    auto message = throwable.message;
+    cause.setProperty(runtime, "name", name);
+    cause.setProperty(runtime, "message", message);
```

Comment:
Inline fields to reduce copies

```suggestion
    cause.setProperty(runtime, "name", throwable.name);
    cause.setProperty(runtime, "message", throwable.message);
```